splunk/scripts/log_indexing.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+logs_path="$SPLUNK_HOME/test_logs"
+endpoint="https://localhost:8089/services/data/inputs/oneshot"
+SPLUNK_PASSWORD="Password"
+
+echo "Indexing logs..."
+for file in "$logs_path"/*; do
+  echo "Indexing $file"
+  curl  --insecure --user admin:"$SPLUNK_PASSWORD" "$endpoint" --data name="$file" --data index="test_index"
+done
+
+if [[ $(curl --insecure --user admin:"$SPLUNK_PASSWORD" "$endpoint" --get --data output_mode=json | jq -r --exit-status 'if .paging.total == 0 then 0 else 1 end') != 0 ]]; then
```

Comment:
breaking lines for readability is not illegal...

Also:
```
if [[ $(foo) != 0 ]]; then ...
```
checks that `foo` did not **print** 0, to check it did not **return** 0 you need to:
```
if foo; then ...
```

You already used `--exit-status` for `jq` which means it would return `0` if its internal calculation results in `true`, but you're not using that.

Reply:
I tried a different variation with ```if ! curl --insecure --user admin:"$SPLUNK_PASSWORD" "$endpoint" --get --data output_mode=json \
  | jq -e '.paging.total == 0'
  ```
  but it didn't work as expected. will see why and fix.